### PR TITLE
chore(aggregations): use same component for output stage preview COMPASS-6451

### PR DESCRIPTION
**Pull request by mabaasit**
### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-aggregations/src/components/stage-preview/output-stage-preview.tsx [hunk 1]
```diff
@@ -85,60 +87,10 @@ export const MergeStage = ({
   onRunOutputStage,
   onGoToOutputResults,
 }: OutputStageProps) => {
-  if (isLoading) {
-    return <Loader destinationNamespace={destinationNamespace} />;
-  }
-
-  // Stage editor show the error message.
-  if (hasServerError) {
+  if (!['$out', '$merge'].includes(operator || '')) {
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/components/stage-preview/output-stage-preview.tsx:90]: We can probably use a helper that we use in other places

```suggestion
  if (isOutputStage(operator ?? '')) {
```

File: packages/compass-aggregations/src/components/stage-preview/output-stage-preview.tsx [hunk 2]
```diff
@@ -214,6 +169,6 @@ const mapDispatch = (
   onGoToOutputResults: () => dispatch(viewOutResults(ownProps.index)),
 });
 
-export const MergeStagePreivew = connect(mapState, mapDispatch)(MergeStage);
+export const MergeStagePreivew = connect(mapState, mapDispatch)(OutputStage);
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/components/stage-preview/output-stage-preview.tsx:172]: Why we are keeping these two as separate exports if they are the same?

File: packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts [hunk 3]
```diff
@@ -316,7 +316,7 @@ describe('Collection aggregations tab', function () {
 
   it('shows $merge stage preview', async function () {
     // $merge operator is supported from 4.4
```
[the remark this answers is not in the document]
Comment (mabaasit) [reply packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts]: yes, thanks. type on comment.

File: packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts [hunk 4]
```diff
@@ -293,16 +293,45 @@ describe('Collection aggregations tab', function () {
...
+
+    expect(text).to.include('Documents will be saved to test.listings.');
+    expect(text).to.include(
+      'The $out operator will cause the pipeline to persist the results to the specified location (collection, S3, or Atlas). If the collection exists it will be replaced.'
+    );
+  });
+
+  it('shows $merge stage preview', async function () {
+    // $merge operator is supported from 4.2.0
+    if (semver.lt(MONGODB_VERSION, '4.2.0')) {
```
Comment (gribnoysup) [inline packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts:319]: Just added a new helper for this:

```suggestion
    if (serverSatisfies('< 4.2.0')) {
```

**Review by gribnoysup — APPROVED** [review]

**Pull request merged**